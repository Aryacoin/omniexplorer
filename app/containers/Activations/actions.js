/*
 *
 * Activations actions
 *
 */

import {
  LOAD_ACTIVATIONS,
  LOAD_ACTIVATIONS_SUCCESS,
} from './constants';

/**
 * Load activations, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_ACTIVATIONS
 */
export function loadActivations() {
  return {
    type: LOAD_ACTIVATIONS,
  };
}
