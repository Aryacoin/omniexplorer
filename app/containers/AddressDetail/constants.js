/*
 *
 * AddressDetail constants
 *
 */

export const DEFAULT_ACTION = 'app/AddressDetail/DEFAULT_ACTION';
