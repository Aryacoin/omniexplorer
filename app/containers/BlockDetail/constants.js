/*
 *
 * BlockDetail constants
 *
 */

export const LOAD_BLOCK = 'omniexplorer/App/LOAD_BLOCK';
export const LOAD_BLOCK_SUCCESS = 'omniexplorer/App/LOAD_BLOCK_SUCCESS';
