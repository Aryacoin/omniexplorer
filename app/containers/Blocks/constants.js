/*
 *
 * Blocks constants
 *
 */

export const LOAD_BLOCKS = 'omniexplorer/App/LOAD_BLOCKS';
export const LOAD_BLOCKS_SUCCESS = 'omniexplorer/App/LOAD_BLOCKS_SUCCESS';
export const LOAD_BLOCKS_ERROR = 'omniexplorer/App/LOAD_BLOCKS_ERROR';
