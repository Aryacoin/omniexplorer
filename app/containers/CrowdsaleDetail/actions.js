import {
  LOAD_CROWDSALE_TRANSACTIONS,
  LOAD_CROWDSALE_TRANSACTIONS_SUCCESS,
  LOAD_CROWDSALE_TRANSACTIONS_ERROR,
  SET_CROWDSALES_TRANSACTIONS_PAGE,
} from './constants';

/**
 * Log an error if the async call fails
 * @param {object} error - the error thrown.
 */
export const errorCrowdsaleTransactionsFetch = error => ({
  type: LOAD_CROWDSALE_TRANSACTIONS_ERROR,
  error,
});

// Start the fetch, toggle is `isFetching` value
export const startCrowdsaleTransactionsFetch = crowdsaleId => ({
  type: LOAD_CROWDSALE_TRANSACTIONS,
  id: crowdsaleId,
});

/**
 * Resolve the fetch with the returned data
 * @param {object} payload - the data returned from the fetch
 */
export const updateCrowdsaleTransactionsFetch = (
  transactions,
  pages,
  total,
  page,
) => ({
  type: LOAD_CROWDSALE_TRANSACTIONS_SUCCESS,
  transactions,
  pages,
  total,
  page,
});

/**
 * Dispatched when the page change
 *
 * @param  {number} page The page number
 *
 * @return {object} An action object with the page
 */
export function setPage(page) {
  return {
    type: SET_CROWDSALES_TRANSACTIONS_PAGE,
    page,
  };
}
