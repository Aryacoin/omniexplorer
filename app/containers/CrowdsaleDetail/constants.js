/*
 *
 * CrowdsaleDetail constants
 *
 */

export const DEFAULT_ACTION = 'app/CrowdsaleDetail/DEFAULT_ACTION';
