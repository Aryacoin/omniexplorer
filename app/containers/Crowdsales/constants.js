/*
 *
 * Search constants
 *
 */

export const LOAD_CROWDSALES = 'omniexplorer/App/LOAD_CROWDSALES';
export const LOAD_CROWDSALES_SUCCESS = 'omniexplorer/App/LOAD_CROWDSALES_SUCCESS';
export const LOAD_CROWDSALES_ERROR = 'omniexplorer/App/LOAD_CROWDSALES_ERROR';
export const LOAD_CROWDSALES_ECOSYSTEM_PROD = 1;
export const LOAD_CROWDSALES_ECOSYSTEM_TEST = 2;

