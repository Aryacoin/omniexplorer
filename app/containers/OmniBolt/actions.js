/*
 *
 * OmniBolt actions
 *
 */
import {
  LOAD_NODES,
  LOAD_USERS,
  LOAD_CHANNELS,
  LOAD_NODES_SUCCESS,
  LOAD_USERS_SUCCESS,
  LOAD_CHANNELS_SUCCESS,
} from 'containers/OmniBolt/constants';

/**
 * Load the nodes, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_NODES
 */
export function loadNodes() {
  return {
    type: LOAD_NODES,
  };
}

/**
 * Load the users, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_NODES
 */
export function loadUsers() {
  return {
    type: LOAD_USERS,
  };
}

/**
 * Load the channels, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_CHANNELS
 */
export function loadChannels() {
  return {
    type: LOAD_CHANNELS,
  };
}

/**
 * Dispatched when the nodes are loaded by the request saga
 *
 * @param  {array} The nodes data
 *
 * @return {object} An action object with a type of FN_API_URL_OMNIBOLT_NODES passing the nodes
 */
export function nodesLoaded(payload) {
  return {
    type: LOAD_NODES_SUCCESS,
    payload,
  };
}

/**
 * Dispatched when the users are loaded by the request saga
 *
 * @param  {array} The users data
 *
 * @return {object} An action object with a type of LOAD_USERS_SUCCESS passing the users
 */
export function usersLoaded(payload) {
  return {
    type: LOAD_USERS_SUCCESS,
    payload,
  };
}

/**
 * Dispatched when the channels are loaded by the request saga
 *
 * @param  {array} The channels data
 *
 * @return {object} An action object with a type of LOAD_CHANNELS_SUCCESS passing the channels
 */
export function channelsLoaded(payload) {
  return {
    type: LOAD_CHANNELS_SUCCESS,
    payload,
  };
}
