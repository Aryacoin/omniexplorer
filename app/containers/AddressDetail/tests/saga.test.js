/**
 * Tests for AddressDetail sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_ADDRESS } from '../constants';
import { addressLoaded, addressLoadingError } from '../actions';

import root, { getAddress } from '../saga';

const addr = '17ScKNXo4cL8DyfWfcCWu1uJySQuJm7iKx';

/* eslint-disable redux-saga/yield-effects */
describe('getAddress Saga', () => {
  let getAddressGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getAddressGenerator = getAddress();

    const selectDescriptor = getAddressGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getAddressGenerator.next(addr).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the addressLoaded action if it requests the data successfully', () => {
    const response = {
      balance: [
        {
          divisible: true,
          frozen: '0',
          id: '31',
          pendingneg: '-6586595182600',
          pendingpos: '0',
          reserved: '0',
          symbol: 'SP31',
          value: '321623187160442',
        },
        {
          divisible: true,
          error: false,
          id: 0,
          pendingneg: '0',
          pendingpos: '0',
          symbol: 'BTC',
          value: '251440000',
        },
      ],
    };

    const putDescriptor = getAddressGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(addressLoaded(response, addr)));
  });

  it('should call the addressLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getAddressGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(addressLoadingError(response)));
  });
});

describe('githubDataSaga Saga', () => {
  const rootSaga = root();

  it('should start task to watch for LOAD_ADDRESS action', () => {
    const takeLatestDescriptor = rootSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_ADDRESS, getAddress));
  });
});
