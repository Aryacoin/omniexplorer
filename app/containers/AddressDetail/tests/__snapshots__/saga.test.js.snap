// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAddress Saga should call the addressLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.omniexplorer.info/v1/address/addr",
      Object {
        "body": "addr=17ScKNXo4cL8DyfWfcCWu1uJySQuJm7iKx",
        "headers": Object {
          "Content-Type": "application/x-www-form-urlencoded",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getAddress Saga should call the addressLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "address": "17ScKNXo4cL8DyfWfcCWu1uJySQuJm7iKx",
      "type": "omniexplorer/App/LOAD_ADDRESS_SUCCESS",
    },
    "channel": null,
  },
}
`;

exports[`getAddress Saga should dispatch the addressLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.omniexplorer.info/v1/address/addr",
      Object {
        "body": "addr=17ScKNXo4cL8DyfWfcCWu1uJySQuJm7iKx",
        "headers": Object {
          "Content-Type": "application/x-www-form-urlencoded",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getAddress Saga should dispatch the addressLoaded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "address": "17ScKNXo4cL8DyfWfcCWu1uJySQuJm7iKx",
      "type": "omniexplorer/App/LOAD_ADDRESS_SUCCESS",
    },
    "channel": null,
  },
}
`;
